File: r/raindrops/raindrops.R
raindrops <- function(number) {
  pling <- if (number %% 3 == 0) "Pling" else ""
  plang <- if (number %% 5 == 0) "Plang" else ""
  plong <- if (number %% 7 == 0) "Plong" else ""

  sound <- paste(pling, plang, plong, sep = "")
  if (sound == "") paste(number) else sound
}

File: ruby/raindrops/raindrops.rb
class Raindrops

  def self.convert(n)
    output = ''
    output << 'Pling' if (n % 3).empty?
    output << 'Plang' if (n % 5).empty?
    output << 'Plong' if (n % 7).empty?

    output == '' ? n.to_s : output
  end

end
